### PR TITLE
onboard centos stream 8 and 9

### PR DESCRIPTION
**Pull request by ksimon1**
This PR adds support for centos stream 8 and 9

```release-note
add support for centos stream 8 and 9 
```
Signed-off-by: Karel Šimon <ksimon@redhat.com>

### CODE REVIEW

File: artifacts/centosstream/centos-stream.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+			DownloadURL:          baseURL + candidate,
+			AdditionalUniqueTags: additionalTags,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("file %q does not exist in the sha256sum file: %v", c.Variant, err)
+
+}
+
+// New accepts CentOS 7 and 8 versions. Example patterns are 7-2111, 7-2009, 8.3, 8.4, ...
```
Comment (dominikholler) [inline artifacts/centosstream/centos-stream.go:91]: This comment should be adjusted to centos stream, e.g.
```
// New accepts CentOS Stream 8 and 9 versions.
```

File: cmd/medius/common/registry.go [hunk 2]
```diff
@@ -27,6 +28,14 @@ var Registry = []Entry{
 		Artifact:   centos.New("8.4"),
 		UseForDocs: true,
 	},
+	{
+		Artifact:   centosstream.New("8"),
+		UseForDocs: true,
```
Comment (dominikholler) [inline cmd/medius/common/registry.go:33]: We want to use only centos 9 in docs.

```
UseForDocs: false,
```

File: cmd/medius/common/registry.go [hunk 3]
```diff
@@ -27,6 +28,14 @@ var Registry = []Entry{
...
+	},
+	{
+		Artifact:   centosstream.New("9"),
```
Comment (dominikholler) [inline cmd/medius/common/registry.go:36]: Maybe the order of ... centos 8, centos 7, centos-stream 9, centos streram 8 is more intuitive.

**Review by dominikholler — CHANGES_REQUESTED** [review]
@ksimon1 thanks for the PR, looks good, the resulting docs looks fine and the produced images works. Would be nice if you could address the three nits.

**Review by dominikholler — APPROVED** [review]
/lgtm

**Comment by dominikholler** [comment]
only errors like "error introspecting image \"quay.io/containerdisks/centos-stream:9\": error parsing image: reading manifest 9 in quay.io/containerdisks/centos-stream: unauthorized: access to the requested resource is not authorized" , which looks fine to me.

File: artifacts/centosstream/centos-stream.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		Description:             description,
+		ExampleCloudInitPayload: docs.CloudInit(),
+	}
+}
+
+func (c *centos) Inspect() (*api.ArtifactDetails, error) {
+	var baseURL string
+
+	if strings.HasPrefix(c.Version, "8") || strings.HasPrefix(c.Version, "9") {
+		baseURL = fmt.Sprintf("https://cloud.centos.org/centos/%s-stream/x86_64/images/", c.Version)
```
Comment (rmohr) [inline artifacts/centosstream/centos-stream.go:40]: `c.Arch` could be used here.

File: artifacts/centosstream/centos-stream.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+	if len(candidates) == 0 {
+		return nil, fmt.Errorf("no candidates for version %q and variant %q found", c.Version, c.Variant)
+	}
+
+	sort.Strings(candidates)
+	candidate := candidates[len(candidates)-1]
+
+	var additionalTags []string
+	additionalTag := strings.TrimSuffix(strings.TrimPrefix(candidate, fmt.Sprintf("CentOS-Stream-%s-", c.Variant)), ".x86_64.qcow2")
```
Comment (rmohr) [inline artifacts/centosstream/centos-stream.go:72]: `c.Arch` could be used here.

**Review by rmohr — APPROVED** [review]
Two changes could be made which apply to the containerdisk resolvers which I wrote too. We can do that in a follow-up.

**Comment by rmohr** [comment]
@ksimon1 @dominikholler looks great :+1:

**Pull request merged**